### DISCUSSION TITLE
ES Lint error causing warning and Netlify failed deployment - app still working fine

### DISCUSSION THREAD

**Discussion by lindseysatterfield** [nss-evening-cohort-14/evening-client-side#296] [Capstones!] [answered]
#### Describe your issue
ES Lint and Netlify do not like my logic for my `handleCheckboxChange` in `RestaurantForm.js`. The app works just fine, but because of the warning, every time I submit a restaurant, a warning pops up in the console. I also think it is messing with my Netlify deployment and failing it for that reason.

Here is my logic that it does not like: 
`[e.target.name]: e.target.name === 'favorite' || 'visited' ? e.target.checked : e.target.value`
Specifically where I get a yellow underline is: 
`e.target.name === 'favorite' || 'visited'`

#### List any error messages that appear
`src/components/RestaurantForm.js
  Line 47:24:  Unexpected constant condition  no-constant-condition`

#### What have you googled? 
https://eslint.org/docs/rules/no-constant-condition
https://stackoverflow.com/questions/37466508/why-does-eslint-trigger-lint-errors-on-whiletrue-using-fibers
https://www.codegrepper.com/code-examples/whatever/unexpected+constant+condition+no-constant-condition

#### What have you tried? Tell us everything you have tried.
- Tried doing an `if` statement instead of a ternary
- Tried moving things on different lines

#### Link to code
Line 47:
https://github.com/lindseysatterfield/nashville-night-out/blob/main/src/components/RestaurantForm.js

**Comment by drteresavasquez** [comment, marked as answer]
You have to specify again event.target.name === “visited” or wrap a parenthesis around favorited || visited. Try that and see what happens.

**Reply by lindseysatterfield** [reply]
Solution: 
`[e.target.name]: e.target.name === 'favorite' || e.target.name === 'visited' ? e.target.checked : e.target.value`
